cpp/ql/test/experimental/query-tests/Security/CWE/CWE-691/semmle/tests/test.c

```diff
@@ -0,0 +1,28 @@
+int tmpFunction(){
+  return 5;
+}
+void workFunction_0(char *s) {
+  int intSize;
+  char buf[80];
+  if(intSize>0 && intSize<80 && memset(buf,0,intSize)) return; // GOOD
+  if(intSize>0 & intSize<80 & memset(buf,0,intSize)) return; // BAD
+  if(intSize>0 && tmpFunction()) return;
+  if(intSize<0 & tmpFunction()) return; // BAD
+}
+void workFunction_1(char *s) {
+  int intA,intB;
+
+  if(intA + intB) return; // BAD
+  if(intA + intB>4) return; // GOOD
+  if(intA>0 && (intA + intB)) return; // BAD
```

Comment:
Similar to the comment on line 15: Is this really bad? If so, then the comment should say `// BAD [NOT DETECTED]`. If it's not actually bad (which I don't think it is?), the comment should say `// GOOD`.